### DISCUSSION TITLE
BSC Network Switch not working Wallet Connect in MetaMask mobile.

### DISCUSSION THREAD

**Discussion by Sudhanmanoharan** [WalletConnect/walletconnect-monorepo#917] [Developer Support]
Hi Team,
    Right now I'm using on wallet connect to connect DAPP,  When I try to switch the BSC network in Metamask mobile App. I’m getting the following error 
**`Error: call revert exception CALL_EXCEPTION [https://docs.ethers.io/v5/troubleshooting/errors/#help-CALL_EXCEPTION].`** 
Then I console Contract, I’m able to get all methods in the console and checked the Network setting, everything looks fine in Metamask mobile for the **BSC network**. But, the same scenario works fine for the **Ethereum network**.  
---
![photo6224015715304845983](https://user-images.githubusercontent.com/27803356/162146867-c471c766-f435-46ae-a131-580982b0068f.jpg)

**Comment by lijiahui94** [comment]
I implemented it in code. I don't know if it helps you
`const network = {
        chainId: 97,
        name: "Binance Smart Chain Testnet",
        symbol: "tBNB",
        rpc: "https://endpoints.omniatech.io/v1/bsc/testnet/public",
    };
    connector.sendCustomRequest({
        method: "wallet_addEthereumChain",
        params: [
            {
                chainId: "0x" + network.chainId.toString(16),
                chainName: network.name,
                nativeCurrency: {
                    name: network.symbol,
                    symbol: network.symbol,
                    decimals: 18,
                },
                rpcUrls: [network.rpc],
            },
        ],
    });`
